File: echoBack/echoBack.c
#include <stdio.h>
#include <string.h>

#define MAXLENGTH 8
char inString[MAXLENGTH];

void main(){
	/* Get string */
	printf("Enter string: ");
	scanf("%s", inString);
	while(1){
		/* Test string */
		if(strlen(inString) > MAXLENGTH){
			// Entered string is too long
			printf("Exceeded max string length [%d]\nEnter string: ", MAXLENGTH);
			scanf("%s", inString);
		} else {
			printf("[ Entered: %s ]\n", inString);
			break;
		}
	}
}

File: 0-helloWorld/echoBack.cs
using System.Runtime.CompilerServices;

namespace echoBack {
    class EchoBack {
        static void Main(string[] args)
        {
            /* Get string */
            int MAXSTRINGLENGTH = 8;
            string inString;
            Console.WriteLine("Enter string: ");
            inString = Console.ReadLine();
            if (!string.IsNullOrEmpty(inString)) { 
                while (true) {
                    /* Test string */
                    if (inString.Length > MAXSTRINGLENGTH) {
                        // Entered string is too long
                        Console.WriteLine("Exceeded max string length [{0}]\nEnter string: ", MAXSTRINGLENGTH);
                        inString = Console.ReadLine();
                    } else {
                        Console.WriteLine("[ Entered: {0} ]", inString);
                        break;
                    }
                }
            }
        }
    }
}